script/tool/lib/src/common/xcode.dart

```diff
@@ -156,4 +166,52 @@ class Xcode {
     }
     return null;
   }
+
+  /// Finds and copies macOS entitlements file. In the copy, disables sandboxing.
+  /// If entitlements file is not found, returns null.
+  ///
+  /// As of macOS 14, testing a macOS sandbox app may prompt the user to grant
+  /// access to the app. To workaround this in CI, we create and use a entitlements
+  /// file with sandboxing disabled. See
+  /// https://developer.apple.com/documentation/security/app_sandbox/accessing_files_from_the_macos_app_sandbox.
+  File? _createDisabledSandboxEntitlementFile(
+    Directory macosDirectory,
+    String configuration,
+  ) {
+    String entitlementDefaultFileName;
+    if (configuration == 'Release') {
+      entitlementDefaultFileName = 'Release';
+    } else {
+      entitlementDefaultFileName = 'DebugProfile';
+    }
```

Comment:
Prefer conditionals for things like this to allow for a single final assignment and no chance of divergent logic:

`final String entitlementDefaultFileName = configuration == 'Release' ? 'Release' : 'DebugProfile';`